tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -2732,6 +2755,23 @@ SpirvVariable *DeclResultIdMapper::createSpirvStageVar(
     return spvBuilder.addStageBuiltinVar(type, sc, BuiltIn::FullyCoveredEXT,
                                          isPrecise, srcLoc);
   }
+  // According to DXIL spec, the ShadingRate SV can only be used by GSOut,
+  // VSOut, or PSIn. According to Vulkan spec, the FragSizeEXT BuiltIn can only
+  // be used as PSIn.
+  case hlsl::Semantic::Kind::ShadingRate: {
+    switch (sigPointKind) {
+    case hlsl::SigPoint::Kind::PSIn:
+      stageVar->setIsSpirvBuiltin();
+      return spvBuilder.addStageBuiltinVar(type, sc, BuiltIn::FragSizeEXT,
+                                           isPrecise, srcLoc);
+    default:
+      emitError("semantic ShadingRate currently unsupported in non-PS shader"
```

Comment:
Actually I just realized the vk.shading-rate.vs-error.hlsl test is not printing this message. (maybe because that test is using VSIn?)
Please also add a test that exercises this code. Thanks

Reply:
Nice catch. Pushed a fix for this.

Reply:
also realized this prints `shaderstages` rather than `shader stages`.